### ISSUE TITLE
"Retry as Root" doesn't make any changes to the document.

### ISSUE THREAD

**Issue by damian101** [notepadqq/notepadqq#923]
I recently noticed that the "Retry as Root" button doesn't work for me anymore. 

**To Reproduce**
1. Open write protected text file with Notepadqq.
2. edit it and clicksave.
3. "Error trying to write to [...]" appears, click on "Retry as Root".
4. Window closes, nothing happens.
5. If you reload document from disk you see that nothing has changed.

**Expected behavior**
After you click "Retry as Root" you get asked for the root password and the changes you made get saved to the document.

**Environment (please complete the following information):**
 - OS: RebornOS (Arch Linux)
 - community repository
 - Qt version 5.13.2
 - Notepadqq version 1.4.8

**Additional context**
gksu is installed. If it is not installed the option "Retry as Root" isn't shown.

**Comment by AlexFolland** [comment]
This issue is still relevant.  I have experienced it also, on Manjaro running Notepad 1.4.8 and Qt version 5.15.0.

**Comment by MaxGodTier** [comment]
This bug was reported on 19 Nov 2019 and it's still here.

I have the same problem as op, but here comes the funny part:

```
sudo notepadqq
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
"Notepadqq will ask for root privileges whenever they are needed if either 'kdesu' or 'gksu' are installed. Running Notepadqq as root is not recommended. Use --allow-root if you really want to."
```

So I'm like "Okay, I'm on a virtual machine, I'll take the risk"

```
sudo notepadqq --allow-root
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
[2952:2952:0817/164035.929035:ERROR:zygote_host_impl_linux.cc(89)] Running as root without --no-sandbox is not supported. See https://crbug.com/638180
```

"Okay, let's do it then."

```
sudo notepadqq --allow-root --no-sandbox
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-root'
Unknown option 'no-sandbox'.
```

Are you kidding me? Gotta love fixing all day so-called "free software" (implying time isn't worth A LOT of money), this is me coming from a hard day where I've already patched a couple dozens of other bugs due to developers negligence but this takes the crown, no offense intended I'm just really tired of this circus.

Environment:

    OS: Ubuntu 20.04 under WSL2
    community repository
    Qt version 5.12.8
    Notepadqq version 2.0.0-beta+git

**Comment by AlexFolland** [comment]
Thank you for reporting the details, and frustration is understandable, but please don't abuse the authors of free software.  They have no obligations, and have only volunteered their time.  I can guarantee much more time was spent on making Notepadqq overall than you've spent on this particular issue.

That being said, `--no-sandbox` is not a notepadqq option.  It seems to be some sort of Chromium option, according to [the URL that the error message included](https://crbug.com/638180).  Are you trying to run Notepadqq in some sort of Chromium-based environment?

**Comment by damian101** [comment]
I can't reproduce this behavior. I'm using Notepadqq 1.4.8 on RebornOS (Arch Linux). Qt version 5.15.2.
Since it's the same version as when I originally created this issue, I assume it's not an inherent issue with Notepadqq...

Can you reproduce it @AlexFolland ? If you're still on Manjaro it should work...

**Comment by AlexFolland** [comment]
I made a new file in my home directory as root called "root file".  I then tried to save it in notepadqq.  I got the "Error trying to write to '/home/alex/root file'" prompt.

![image](https://user-images.githubusercontent.com/601517/129835731-11e1ca62-6900-4892-8f13-8c87b9bfef3c.png)

I pressed "Retry as Root".  I got the KDE su prompt to type the root password, showing the exact command that notepadqq was trying to execute.

![image](https://user-images.githubusercontent.com/601517/129835807-f9606063-6ad1-4ee3-b3be-29a9982bcdb4.png)

I typed the root password and pressed "OK".  I closed notepadqq and viewed "/home/alex/root file" again.  My changes were intact.

I suspect the thing that changed on my system is the fact that I have `kdesu` installed now, whereas I didn't before.  I have recently been experimenting with trying to get KDE Plasma to work alongside XFCE, and `kdesu` is one of the dependencies.  So, maybe this works now because of that.  I'm not sure though.

Either way, for me, this is no longer an issue.  I hope this helps.
